docs/contribution-guides/Development-Standards-Guide.md

```diff
@@ -226,3 +242,301 @@ to just leave it as a public field without a property. As always, make sure to d
+
+Unity has a great talk on how to do just that in this video: [How to adopt a modular code workflow for your team | Unite 2022](https://youtu.be/nk3gHIZZ5Rg)
+
+Below are good practices and explainations on what to do, so you can avoid having your code refactored/reworked 5 months later and make the lives of other contributors much easier.
+
+## 0. In-code documentation
+
+* You can avoid confusing others (and even yourself) by naming your functions and variables in a clear and descriptive way.
+
+```cs
+//bad code
+void doThing(bool b)
+{
+    state = b;
+}
+```
+
+```cs
+//good code
+private void ThisFunctionWillDoA_B_And_C(bool newState)
+{
+    state = newState;
+}
+```
+
+* Explain what your function does step by step whenever possible. Your code might seem obvious to you, but for other people; that may not be the case.
+
+```cs
+//Example from NightVisionGoggle.cs
+
+/// <summary>
+/// Checks if the item is in the correct ItemSlot which is the eyes.
+/// Automatically returns false if null because of the "is" keyword and null propagation.
+/// </summary>
+private bool IsInCorrectNamedSlot()
+{
+	return pickupable.ItemSlot is { NamedSlot: NamedSlot.eyes };
+}
+
+private void ApplyEffects(bool state)
+{
+	var finalState = state;
+	// If for whatever reason unity is unable to catch the correct main camera that has the CameraEffectControlScript
+	// Don't do anything.
+	if (Camera.main == null ||
+		Camera.main.TryGetComponent<CameraEffectControlScript>(out var effects) == false) return;
+	// If the item is not in the correct slot, ensure the effect is disabled.
+	if (IsInCorrectNamedSlot() == false) finalState = false;
+	// Visibility is updated based on the on/off state of the goggles.
+	// True means its on and will show an expanded view in the dark by changing the player's light view.
+	// False will revert it to default.
+	effects.AdjustPlayerVisibility(finalState);
+}
+```
+
+
+* Make proprer use of `Logger`
+
+It's easy to debug problems when the game tells us that there is one. As a result, we use `Logger`; a logging system that allows us to store records of how the game has been behaving/misbehaving while it's running.
+
+Logger should only be used to report unintended behaviors, errors and backend system messages. Never spam logger and remove any debugging messages you have left once you're done using it to print testing values/results in the console.
+
+To make our lives easier as well, add the class/gameObject name and function in the log message so it's easier to tell ourselves what went wrong from a surface.
+
+_Note: Avoid using unity's `Debug` class for logging as it doesn't store logs where we need them to be._
+```cs
+//bad use of logger.
+
+Logger.Log("something happened");
+```
+```cs
+//good use of logger.
```

Comment:
The examples of good use arent using the logger categories

Reply:
I honestly don't even know whats the point of categories, it just seems like a way to silently hide logs since most of the logs under categories aren't even visible to the client.

Reply:
Don't they get written to the log file? Categories aren't meant for the client.

Reply:
Logger prints to the console and discord, if it's hidden behind a category it wouldn't alarm anyone about it, and we'd need to go through long log files to find what we need.
Ask yourself this, when was the last time we, as the developers, looked at a log file directly or asked one of our players to send their entire log file?

Reply:
When you are only interested in reading a category, it is good to have the option.

Reading logs on discord is a misbehaviour of us, that really needs to change.

Reply:
Alright then, will update to change it.